rasa/shared/nlu/training_data/message.py

```diff
@@ -398,3 +398,18 @@ def is_e2e_message(self) -> bool:
             (self.get(ACTION_TEXT) and not self.get(ACTION_NAME))
             or (self.get(TEXT) and not self.get(INTENT))
         )
+
+    def find_overlapping_entities(
+        self,
+    ) -> List[Tuple[Dict[Text, Any], Dict[Text, Any]]]:
+        """Finds any overlapping entity annotations."""
+        entities = self.get("entities", [])[:]
+        entities.sort(key=lambda e: e["start"])
+        overlapping_pairs: List[Tuple[Dict[Text, Any], Dict[Text, Any]]] = []
+        for i, entity in enumerate(entities):
+            for other_entity in entities[i + 1 :]:
+                if other_entity["start"] < entity["end"]:
+                    overlapping_pairs.append((entity, other_entity))
+                else:
+                    break
```

Comment:
It actually collects all overlapping pairs. The break is here since the list is ordered: once you find an `other_entity` which starts after your `entity` ends, there can't be any `other_entity` which overlaps with `entity` anymore. So you move on to the next  `entity`.

Reply:
Makes sense, thanks 🙂